daffodil-cli/src/it/scala/org/apache/daffodil/CLI/Util.scala

```diff
@@ -200,4 +203,25 @@ object Util {
     }
     inputFile
   }
+
+  def expectExitCode(expectedExitCode: ExitCode.Value, shell: Expect): Unit = {
+    val expectedCode = expectedExitCode.id
+
+    val keyWord = "EXITCODE:"
+    val exitCodeCmd = "echo " + keyWord + (if (Util.isWindows) "%errorlevel%" else "$?")
+    shell.sendLine(exitCodeCmd)
+    shell.expect(contains(keyWord))
```

Comment:
It looks like a handful of tests are failing to find this exit code, any idea why that is?

Reply:
from running the tests manually i've seen the Expect operation fails due to timeout

Reply:
Is there a race condition between what we're telling it to echo back, and stuff it was going to output anyway?

Reply:
Looking at a handful of failing tests, it looks like ones that have this issue are ones that start the shell with:
```scala
val shell = Util.start("", true)
```
The `true` here is a flag sets the `expectErr` parameter, and looking at the getShell function, it ends up doing this:
```scala
    val inputStream = if (expectErr) {
      process.getErrorStream()
    } else {
      process.getInputStream()
    }
    val shell = new ExpectBuilder()
      .withInputs(inputStream)
      ...
```
I think what this is saying is that when `expectErr = true`, expect only captures the error stream (i.e. stderr) instead of the input stream (i.e. stdout). So when `Util.start("", true)`, expect will never see EXITCODE because that is always output to stdout, which isn't captured. So it's impossible to use `expectErr = true` and check the exitcode.

Since all integration tests now need access to stdout to check against the exit code, that suggests that we really can't use the `Util.start("", true)` function anymore to only capture stderr. We always need to capture stdout too.

So, I wonder if our integration tests should just always capture BOTH stdout and stderr, and when they want to check something in stderr, they need to be explicit about it?

Note that we already have a function that does this: `Util.sartIncludeErrors`. The difference here is if you want to expect something in stderr, you need to be explicit and do this:
```scala
shell.expectIn(1, ...)
```
The `1` here says to check stderr.

So I'm thinking maybe that suggests we should just get rid of `getShell` and rename `getShellWithErrors` to `getShell`. That way all tests will capture both stdout and stderr. And it means `expectErr` parameter just goes away, and all tests are started with `Util.start("")`. And then we need to update tests that are looking at stderr (i.e. the ones passing `true` to the `start()` method) to use `shell.expectIn(1, ...)` instead of `shell.expect(...)`. Looks like there's 36 of these tests, so a decent number, but not a ton, and the changes are fairly straightforward.